CMake/lrs_options.cmake

```diff
@@ -35,6 +35,7 @@ option(FORCE_LIBUVC "Explicitly turn-on libuvc backend - deprecated, use FORCE_R
 option(FORCE_WINUSB_UVC "Explicitly turn-on winusb_uvc (for win7) backend - deprecated, use FORCE_RSUSB_BACKEND instead" OFF)
 option(ANDROID_USB_HOST_UVC "Build UVC backend for Android - deprecated, use FORCE_RSUSB_BACKEND instead" OFF)
 option(CHECK_FOR_UPDATES "Checks for versions updates" ON)
+option(BUILD_WITH_CPU_EXTENSIONS "Enable compiler optimizations using CPU extensions (such as AVX)" ON)
```

Comment:
The name is misleading as we still use SSE2/SSE3 in default configuration.
We need the additional granularity level and separate AVX,AVX2(512) and may be SSE3 for X86 architecture.
~~BUILD~~**DISABLE**_CPU_EXTENSIONS="AVX2`  yo disable itnrinsic of AVX2 and above and then apply it accordingly

Reply:
As discussed:
I asked to start with `BUILD_` to keep in line with the other options.
The granularity is too much for right now, and I don't see the need or ROI at this point in time. We'll change as needed.